modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/CreationMenu.js

```diff
@@ -190,7 +197,13 @@ const ItemList = ({primaryItems, secondaryItems, visibleItemsCount}) => {
 				}
 
 				return (
-					<ClayDropDown.Item href={item.href} key={index}>
+					<ClayDropDown.Item
+						href={item.href}
```

Comment:
If we want to support links with an `onClick`, I am ok with this. Otherwise, I think it would be better to be restrictive, by setting only the `href` or the `onClick`

Reply:
That's a very good question @ambrinchaudhary.

I understand your concern and when adding this change I also thought about it.

I think that we could support both `href` and `onClick` since `<ClayDropDown.Item>` (or `<ClayLink>`) supports both, but we can restrict if you think it's better.

Reply:
@julien We have similar logic in [action controls](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ActionControls.js#L40-L67):
```
item.href ? <ClayLink ... /> : <ClayButtonWithIcon ... />
```

We should be able to define `onClick` on both link and button. If `href` is not defined, we show a button. This matches [Clay v2](https://github.com/liferay/clay/blob/2.x/packages/clay-dropdown/src/ClayDropdownItem.soy#L140-L165) and should cover all cases in usages.

[Clay v3](https://github.com/liferay/clay/blob/master/packages/clay-drop-down/src/Item.tsx#L66-L67) also supports a `span`, if there is no `href` and `onClick`. There are no usages with this case, but we can pretty painlessly add support for it. We can just copy the logic from Clay v3. 

Can we use `ClayDropDownItem` directly? :thinking: This would simplify things a lot. So instead of 
```
item.href ? <ClayLink ... /> : <ClayButtonWithIcon ... />
```
we would have 
```
<ClayDropDownItem href={} onClick={} ...  />
```

If not, we can extract the logic into something like `DropdownItemElement.js`, and use it both from this file and from `ActionControls.js`

Reply:
@markocikos, there are somethings I don't understand

> We should be able to define onClick on both link and button

Yes, that's already the case right?

> Can we use ClayDropDownItem directly? thinking This would simplify things a lot. 

Isn't that the case?

Reply:
@julien Yeah, you're right, we already have all that. Sorry, I was looking too much in `ActionControls`, not enough in file in this PR :slightly_smiling_face:. `ActionControls` is the one that could be simplified, with similar logic to `ClayDropDown.Item`. But that's outside of scope of this PR:

All good here :+1: 

I would only consider extracting the item into a private component, so we don't have duplication:
```js
const Item = ({...}) => {
	return (<ClayDropDown.Item ... />);
};
```

Reply:
Hey @markociko no problem, that happens to me too :smile: 

> I would only consider extracting the item into a private component, so we don't have duplication

Sounds like a good idea yes, let me look into that and send another commit (which I'll probably squash before forwarding but I don't want to loose this conversation).